app/(landing)/run-for-office/components/Testimonials.js

```diff
@@ -0,0 +1,238 @@
+    image: {
+      url: '//images.ctfassets.net/g08ybc4r0f4b/5F56YyUMfJViMshnO52CDZ/8f072f7e875deb4d33aa1537efe57899/Mary_Anderson.jpg',
+      alt: 'Mary Anderson',
+    },
+    testimonial:
+      'kdfjskf sldkjf lsdkfj sdlkfj sldkfj lsdkfj sdlkfj sldkf jlsdkfj dlskfj lsdkfj lsdkfj lksjf lksdjf lksdjf lks',
+  },
+  {
+    name: 'Jenny Joni',
+    office: 'Judge forever and ever',
+    image: {
+      url: '//images.ctfassets.net/g08ybc4r0f4b/5aOMolTAK1apiOWJl3Xuvt/d4d4994aa4465365fd45074cc2508be8/Japjeet_Uppal.jpg',
+      alt: 'Japjeet Uppal',
+    },
+    testimonial:
+      'fdslkfjdslk sldkfj sdlkfj sdlkfj lskdfj lskdjf lsdkjf lskdj flksdj flskdjf lksdj flksdj flskd jflksdj flks jf.',
+  },
+  {
+    name: 'Patrick Porkins',
+    office: 'Candidate for SC House District 111',
+    image: {
+      url: '//images.ctfassets.net/g08ybc4r0f4b/7iXcLSYYfoG6KuzALoEuSo/5cae482b9311ad4ac6cbf959f5560c0d/Joe_Jernigan.jpg',
+      alt: 'Joe Jernigan',
+    },
+    testimonial:
+      'fkdjs lkdsjf sdlkfj lksdfj lsdkfj lsdkfj lsdkjf lkdsjf lksdjf ldskfj ldskfj lsdk',
+  },
+  {
+    name: 'Amy Amyson',
+    office: 'Candidate for Snohomish, County WA Judge',
+    image: {
+      url: '//images.ctfassets.net/g08ybc4r0f4b/5F56YyUMfJViMshnO52CDZ/8f072f7e875deb4d33aa1537efe57899/Mary_Anderson.jpg',
+      alt: 'Mary Anderson',
+    },
+    testimonial:
+      'KDFJSDKPFJ S:KDFJ SD:KF JSDKF JSLDKF JLJFKSLDJF LKSDJF KLSJDFLKSDJ FKLSDJ FLKSDJ FLKSDJF KLSDJ FKLDS fjSK',
+  },
+  {
+    name: 'Japjeet Uppal',
+    office: 'Candidate for Livingston, CA City Council',
+    image: {
+      url: '//images.ctfassets.net/g08ybc4r0f4b/5aOMolTAK1apiOWJl3Xuvt/d4d4994aa4465365fd45074cc2508be8/Japjeet_Uppal.jpg',
+      alt: 'Japjeet Uppal',
+    },
+    testimonial:
+      "It's wild to me still that I get to do all of this for free. I very much appreciate GoodParty.org's mission and that there's people like you who are willing to put the time in to invest in folks like me all across the country, because we 100% need it.",
+  },
+];
+
+function PageDot({ pageNum, isSelected, onClick }) {
+  return (
+    <span
+      onClick={onClick}
+      role="button"
+      title={'Page ' + ++pageNum}
+      className={`inline-block rounded-full h-2 w-2 mx-1 ${
+        isSelected ? 'bg-primary-main' : 'bg-black/[0.38]'
+      }`}
+    ></span>
+  );
+}
+
+export default function Testimonials({ testimonials }) {
+  const [currentPage, setCurrentPage] = useState(0);
+  const [pageSize, setPageSize] = useState(SM_PAGE_SIZE);
+
+  useEffect(() => {
+    window.addEventListener('resize', handleResize);
+
+    handleResize();
+
+    return () => window.removeEventListener('resize', handleResize);
+
+    function handleResize() {
+      const windowWidth = window.innerWidth;
+      if (windowWidth <= LG_MIN) {
+        setPageSize(windowWidth <= MD_MIN ? SM_PAGE_SIZE : MD_PAGE_SIZE);
+      } else {
+        setPageSize(LG_PAGE_SIZE);
+      }
+    }
```

Comment:
You would need to? Like, React won't let you use `window` in a method declaration inside of the function body of a functional component? 🤔 

I'm not sure I'm following?

https://codesandbox.io/p/sandbox/angry-jerry-r8rkk4

Reply:
Well.. there's a whole host of reasons why folks don't typically place function declarations within other functions. Some subjective, others objective.  AI gives a pretty good outline here why avoiding this tends to be good practice, if that's what you're asking, I'll paste it below? 🤷

Personally, I tend to fall under the "Performance and Scope" reasons for not doing this. And yes, I understand that we have these all over our components, but not so often within the callback functions of hooks, unless we're needing to define an async behavior.  Even then, it's questionable as to whether or not someone "needs" to follow this pattern to accomplish what they're trying to do. 

Personally, I wish we used more thunks to pull nested  methods up and out of our component function bodies. But that hasn't been my call to make. 🤷 

> While JavaScript allows you to declare functions within functions (nested functions), it's often considered bad practice for a few reasons:
Readability and Maintainability:
Increased complexity:
Nested functions can make your code harder to read and understand, especially as the nesting level increases. This is because you have to keep track of the scope chain to understand where variables are coming from and how they're being used.
Difficult to reuse:
Nested functions are tightly coupled to their parent functions. This makes it harder to extract and reuse them in other parts of your codebase.
Performance and Scope:
Scope chain traversal:
When you call a nested function, JavaScript has to traverse the scope chain to find the variables it needs. This can have a slight performance impact, especially in situations where the nested function is called frequently.
Unintended closures:
Nested functions create closures, which means they can access variables from their parent function even after the parent function has finished executing. This can lead to unintended consequences if you're not careful.
Memory usage:
Each nested function creates a new scope, which consumes memory. In some cases, this can lead to memory leaks if not managed properly.
Alternatives to Nested Functions:
Function declarations outside the parent function:
If you need to reuse a function, declare it outside the parent function so it can be accessed from anywhere in your code.
Module pattern:
Use the module pattern to encapsulate related functionality and avoid polluting the global scope.
Classes (ES6 and later):
If you're working with object-oriented code, use classes to organize your methods and data.
However, there are situations where nested functions can be useful:
Private functions:
If you need a helper function that is only used within the parent function, a nested function can help encapsulate that logic and prevent it from cluttering the global scope.
Closures for data encapsulation:
Closures created by nested functions can be used to encapsulate data and create private variables that can't be accessed from outside the function.
In general, it's best to avoid nested functions unless you have a specific reason to use them. Keep your code simple, readable, and maintainable.

Reply:
Not a hill worth dying on for me. Like I said, there are indeed objective reasons for not doing this.  I'm not asking you to change something based on a subjective opinion of my own.

But again, not worth arguing about in this case.